src/main/kotlin/ArchiveCreationScreen.kt

```diff
@@ -0,0 +1,29 @@
+import java.util.Scanner
+
+object ArchiveCreationScreen : Creatable {
+    override fun createObject() {
+        val listFromHandler = mutableMapOf<String, (() -> Unit)?>()
+        listFromHandler["\nСОЗДАНИЕ АРХИВА:"] = null
+        listFromHandler["\tВведите название создаваемого архива:"] = {
+            ListOfArchives.add(Archive(createArchiveName()))
+        }
+        listFromHandler["\tОперация успешно завершена."] = { ArchiveSelectionScreen.menuOutput() }
+        InputOutputHandler(listFromHandler).inputOutputOperations(this)
+    }
+
+    private fun createArchiveName(): String {
+        var name = ""
+        var nameIsPossible: Boolean
+        do {
+            nameIsPossible = true
+            name = Scanner(System.`in`).nextLine()
+            ListOfArchives.forEach { if (name == it.name) nameIsPossible = false }
+            if (!nameIsPossible || name == "") {
```

Comment:
🍏 Можно сначала выполнить более простую операцию - определить, что имя не пустое. А затем, если подтвердилось, что имя не пустое, то запустить более тяжелую операцию- по поиску имени в списке.  При этом не требуется проходить по всему списку, достаточно найти первое совпадающее имя. Функция find делает именно это.
```suggestion
            if (name.isEmpty || ListOfArchives.find {it.name == name } != null) {
```